tests/test_embeds.py

```diff
@@ -289,6 +289,79 @@ def test_fields_exceptions() -> None:
         embed.set_field_at(0, "x", "y")
 
 
+def test_field_restraints() -> None:
+    embed = Embed(title="T" * 256)  # Size = 256
+    embed.set_footer(text="T" * 2048)  # Size = 2304
+    embed.add_field(name="A" * 256, value="B" * 1024)  # Size = 3584
+    embed.add_field(name="A" * 256, value="B" * 1024)  # Size = 4864
+    embed.add_field(name="A" * 112, value="B" * 1024)  # Size = 6000
+    assert len(embed) == 6000
+    assert embed.check_limits()
+
+    embed.add_field(name="A", value="B")  # Breaks limit of 6000 chars
+    with pytest.raises(ValueError, match="^Embed total size"):
+        embed.check_limits()
+    embed.remove_field(3)
+
+    assert embed.check_limits()
+    embed.insert_field_at(index=3, name="A", value="B")  # Breaks limit of 6000 chars
+    with pytest.raises(ValueError, match="^Embed total size"):
+        embed.check_limits()
+    embed.remove_field(3)
+
+    embed.set_field_at(index=2, name="A" * 113, value="B" * 1024)  # Breaks limit of 6000 chars
+    assert len(embed) == 6001
+    with pytest.raises(ValueError, match="^Embed total size"):
+        embed.check_limits()
+    embed.set_field_at(index=2, name="A" * 112, value="B" * 1024)
+    assert len(embed) == 6000
+
+    embed.set_author(name="A")  # Breaks limit of 6000 chars
+    with pytest.raises(ValueError, match="^Embed total size"):
+        embed.check_limits()
+    embed.remove_author()
+
+    embed.set_footer(
+        text="T" * 2048 + " " * 500
+    )  # Would break the 6000 limit, but leading + trailing whitespace doesn't count
+    assert embed.check_limits()
+
+    embed = Embed(title="Too many fields :WAYTOODANK:")
+    for _ in range(25):
+        embed.add_field(name="OK", value=":D")
+    assert embed.check_limits()
```

Comment:
```suggestion
    embed.check_limits()

    embed.add_field(name="A", value="B")  # Breaks limit of 6000 chars
    with pytest.raises(ValueError, match="^Embed total size"):
        embed.check_limits()
    embed.remove_field(3)

    embed.check_limits()
    embed.insert_field_at(index=3, name="A", value="B")  # Breaks limit of 6000 chars
    with pytest.raises(ValueError, match="^Embed total size"):
        embed.check_limits()
    embed.remove_field(3)

    embed.set_field_at(index=2, name="A" * 113, value="B" * 1024)  # Breaks limit of 6000 chars
    assert len(embed) == 6001
    with pytest.raises(ValueError, match="^Embed total size"):
        embed.check_limits()
    embed.set_field_at(index=2, name="A" * 112, value="B" * 1024)
    assert len(embed) == 6000

    embed.set_author(name="A")  # Breaks limit of 6000 chars
    with pytest.raises(ValueError, match="^Embed total size"):
        embed.check_limits()
    embed.remove_author()

    embed.set_footer(
        text="T" * 2048 + " " * 500
    )  # Would break the 6000 limit, but leading + trailing whitespace doesn't count
    embed.check_limits()

    embed = Embed(title="Too many fields :WAYTOODANK:")
    for _ in range(25):
        embed.add_field(name="OK", value=":D")
    embed.check_limits()
```